app/models/miq_report_result/result_set_operations.rb

```diff
@@ -0,0 +1,43 @@
+module MiqReportResult::ResultSetOperations
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    def apply_limit_and_offset(results, options)
+      results.slice(options['offset'].to_i, options['limit'].to_i) || []
+    end
+
+    def format_result_set(miq_report, result_set)
```

Comment:
> Should this be MiqReport#format_result_set(result_set) ?

> Seems all variables and loops are part of miq_report

Yes, I  moved it to [miq report.](https://github.com/ManageIQ/manageiq/pull/18532/commits/f9e97e974906fc09428017359d7eb074ba636593)

> Also feels like result_set should not be a hash but rather an array (in the correct order). But maybe the way miq_report.format_column is written, it would not be possible.

I think it is not possible because transform values can use method to transform values without parameters and we need some parameters. So at least I extract code to method[ format_row.](https://github.com/ManageIQ/manageiq/pull/18532/commits/18f6c323b1ebc61fc8ac96895387cbd60f2c4d00)